Dockerfile

```diff
@@ -0,0 +1,78 @@
+# Build Stage: Build the application
+FROM rust:latest AS builder
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy the main Cargo.toml and Cargo.lock files
+COPY Cargo.toml Cargo.lock ./
+
+# Copy the Cargo.toml files from each crate
+COPY web/Cargo.toml web/Cargo.toml
+COPY service/Cargo.toml service/Cargo.toml
+COPY entity_api/Cargo.toml entity_api/Cargo.toml
+COPY entity/Cargo.toml entity/Cargo.toml
+COPY migration/Cargo.toml migration/Cargo.toml
+
+# Verify binaries are created
+RUN ls -la /app/
+
+# Fetch dependencies based on the Cargo.toml and Cargo.lock
+RUN cargo fetch 
+
+# Copy the source code
+COPY . .
+
+# Build the application in release mode
+RUN cargo build --release
+
+# Runtime Stage: Create the runtime image
+FROM debian:buster-slim
+
+# Install necessary system dependencies
+RUN apt-get update && apt-get install -y \
+    libssl1.1 \
+    ca-certificates \
+    libpq5 \
+    && rm -rf /var/lib/apt/lists/*
```

Comment:
I'm always curious how one determines exactly what is needed here.

Reply:
I added this part to make sure the container has all the system dependencies needed for the Rust app to run correctly, specifically libraries and certificates. `libssl1.1` for SSL support, ca-certificates for verifying `HTTPS` connections, and `libpq5` for `PostgreSQL` connections. Clearing /var/lib/apt/lists just reduces image size by removing cached package data. Hope that helps.